src/main/java/com/flagsmith/flagengine/utils/types/TypeCasting.java

```diff
@@ -59,7 +88,7 @@ public static Boolean compare(SegmentConditions condition, Comparable value1, Co
   public static Float toFloat(Object number) {
     try {
       return number instanceof Float ? ((Float) number) : Float.parseFloat((String) number);
-    } catch (NumberFormatException nfe) {
+    } catch (Exception nfe) {
```

Comment:
How come we've changed this (and the lines below) to catch any exception?

Reply:
Copying from Slack. The Exceptions thrown at those can be of different types, which is why used the generic Exception interface.

Reply:
Resolving as I've spun this off into a separate issue [here](https://github.com/Flagsmith/flagsmith-java-client/issues/70).